shell/common/variable_refresh_rate_display.cc

```diff
@@ -9,17 +9,25 @@ namespace flutter {
 
 VariableRefreshRateDisplay::VariableRefreshRateDisplay(
     DisplayId display_id,
-    const VariableRefreshRateReporter& refresh_rate_reporter)
-    : Display(display_id, refresh_rate_reporter.GetRefreshRate()),
+    const std::weak_ptr<VariableRefreshRateReporter> refresh_rate_reporter)
```

Comment:
nit: might be better to have a method like this

```
static void GetInitialRefreshRate(const std::weak_ptr<VariableRefreshRateReporter>& refresh_rate_reporter) {
    if (auto reporter = refresh_rate_reporter.lock()) {
        return reporter->GetRefreshRate((;
    } else {
        return 0;
    }
}
```

And use this at both here and in line 21 like so:

`: Display(GetInitialRefreshRate(refresh_rate_reporter))`